app-services/secrets/README.md

```diff
@@ -1,36 +1,112 @@
+
+   ```yml
+     app-service-secrets-example:
+       image: edgexfoundry/docker-secrets-example:dev
+       container_name: app-service-secrets-example
+       hostname: app-service-secrets-example
+       ports:
+         - "48095:48095"
+       networks:
+         - edgex-network
+       environment:
+         <<: *common-variables
+         SecretStore_Protocol: https
+         SecretStore_TokenFile: /tmp/edgex/secrets/edgex-appservice/secrets-token.json
+         SecretStoreExclusive_Host: edgex-vault
+         SecretStoreExclusive_Protocol: https
+         SecretStoreExclusive_RootCaCertPath: /tmp/edgex/secrets/ca/ca.pem
+         SecretStoreExclusive_ServerName: edgex-vault
+         SecretStoreExclusive_TokenFile: /tmp/edgex/secrets/secrets-example/secrets-token.json
+       volumes:
+         - /tmp/edgex/secrets/ca:/tmp/edgex/secrets/ca:ro,z
+         - /tmp/edgex/secrets/edgex-appservice:/tmp/edgex/secrets/edgex-appservice:ro,z 
+         - /tmp/edgex/secrets/secrets-example:/tmp/edgex/secrets/secrets-example:ro,z 
+       depends_on:
+         - vault-worker
+   ```
+
+   4. Run EdgeX security services in docker:
 
-Application Services can retrieve their secrets from the underlying secret store using the GetSecrets() API in the SDK. 
+      The secrets example in docker is dependent on the rest of the EdgeX security services, at a minimum. 
 
-`.GetSecrets(path string, keys ...string)` is used to retrieve secrets from the secret store. `path` describes the type or location of the secrets to retrieve. If specified it is appended to the base path from the secret store configuration. `keys` specifies the secrets which to retrieve. If no keys are provided then all the keys associated with the specified path will be returned.
+      1. Run the secret store services in EdgeX using:
 
-#### Secure mode
+         `docker-compose -f <docker-compose file name> up -d vault-worker`
 
-If in secure mode, the secrets are stored and retrieved from Vault based on the SecretStore configuration values.
+         Verify that the following services are running:
 
-Setup the application config to use  Vault's secrets engine:
+         ```
+         edgex-core-consul
+         edgex-files      
+         edgex-secrets-setup
+         edgex-vault        
+         edgex-vault-worker
+         ```
+
+      2. Wait for about a minute to ensure tokens are created for the services that were specified in the docker compose section of vault-worker service for `ADD_SECRETSTORE_TOKENS`environment variable. Verify that the tokens for `edgex-appservice` and `secrets-example` exist under `/tmp/edgex/secrets`.
+
+   5. Run the secrets example service:
+
+      ```
+      docker-compose -f <docker-compose file name> up -d app-service-secrets-example
+      ```
+
+      Check the app's logs to make sure no errors occurred during startup:
+
+      ```
+      docker logs app-service-secrets-example
```

Comment:
Following the instructions I get the following output.  Just before step 1 you say "The secrets example in docker is dependent on the rest of the EdgeX security services, at a minimum" which implies that I don't need the rest of EdgeX running. I think it would be a good idea to a blockquote that says it is normal to see this message if you aren't running the full EdgeX stack and can safely ignore it. On another note - @lenny-intel Is there way we can silence this error message? And why does it not follow logging standards?

![image](https://user-images.githubusercontent.com/22204562/80146424-ae8f3280-8566-11ea-9c22-5cedfe11c82e.png)

Reply:
@rsdmike , @eno-intel , Yes, removing the Core Data from the configuration clients configuration should stop it.  if not expecting any of those services, then the Client section of the config should be removed. They are optional based on the features you plan on using.

This message is from the code that is pinging Consul for the current endpoint and is in Core Contracts where it was never passed the logging client.

Reply:
Actually, no it is in our own code. Copied from the other repos.
https://github.com/edgexfoundry/app-functions-sdk-go/blob/499fea2e8f95bc135a1d8ddbd827dd9280452983/pkg/endpoint/endpoint.go#L83 

Need to add a tech debt issue to clean this up.

Reply:
opted for removing core-data from the config

Reply:
added tech debt issue - cant label it though.
https://github.com/edgexfoundry/app-functions-sdk-go/issues/342